packages/core/src/locales/id.ts

```diff
@@ -0,0 +1,125 @@
+import type { $ZodStringFormats } from "../checks.js";
+import type * as errors from "../errors.js";
+import * as util from "../util.js";
+
+const Sizable: Record<string, { unit: string; verb: string }> = {
+  string: { unit: "characters", verb: "memiliki" },
+  file: { unit: "byte", verb: "memiliki" },
+  array: { unit: "item", verb: "memiliki" },
+  set: { unit: "item", verb: "memiliki" },
+};
+
+function getSizing(origin: string): { unit: string; verb: string } | null {
+  return Sizable[origin] ?? null;
+}
+
+export const parsedType = (data: any): string => {
+  const t = typeof data;
+
+  switch (t) {
+    case "number": {
+      return Number.isNaN(data) ? "NaN" : "number";
+    }
+    case "object": {
+      if (Array.isArray(data)) {
+        return "array";
+      }
+      if (data === null) {
+        return "null";
+      }
+
+      if (Object.getPrototypeOf(data) !== Object.prototype && data.constructor) {
+        return data.constructor.name;
+      }
+    }
+  }
+  return t;
+};
+
+const Nouns: {
+  [k in $ZodStringFormats | (string & {})]?: string;
+} = {
+  regex: "input",
+  email: "email address",
+  url: "URL",
+  emoji: "emoji",
+  uuid: "UUID",
+  uuidv4: "UUIDv4",
+  uuidv6: "UUIDv6",
+  nanoid: "nanoid",
+  guid: "GUID",
+  cuid: "cuid",
+  cuid2: "cuid2",
+  ulid: "ULID",
+  xid: "XID",
+  ksuid: "KSUID",
+  datetime: "ISO datetime",
+  date: "ISO date",
+  time: "ISO time",
+  duration: "ISO duration",
+  ipv4: "IPv4 address",
+  ipv6: "IPv6 address",
+  cidrv4: "IPv4 range",
+  cidrv6: "IPv6 range",
+  base64: "base64-encoded string",
+  base64url: "base64url-encoded string",
+  json_string: "JSON string",
+  e164: "E.164 number",
```

Comment:
```suggestion
  datetime: "tanggal dan waktu format ISO",
  date: "tanggal format ISO",
  time: "jam format ISO",
  duration: "durasi format ISO",
  ipv4: "alamat IPv4",
  ipv6: "alamat IPv6",
  cidrv4: "rentang alamat IPv4",
  cidrv6: "rentang alamat IPv6",
  base64: "string dengan enkode base64",
  base64url: "string dengan enkode base64url",
  json_string: "string JSON",
  e164: "angka E.164",
```

Need to debate whether to include "format" word for brevity. cc @resir014